identity/identity.py

```diff
@@ -69,6 +69,20 @@ def validate_urn(urn):
         raise ImproperlyConfigured(f"Invalid urn: '{urn}'")
 
 
+def get_validated_claims(token):
+    """
+    Validates JWT and returns the claims if validated, None otherwise.
+    """
+    validated_jwt = jwt.JWT(
+        key=container.platform_jwks, jwt=token
+    )
+    claims = json.loads(validated_jwt.claims)
+    if "use" in claims and claims["use"] == "access" and "sub" in claims:
```

Comment:
Out of curiosity, what does line 80 do?

Reply:
I noticed we just moved this into a separate function in identity (originally code is in `permissions.py`). Curious why you made that decision as well.

Reply:
To my knowledge, this line checks that the use case of the claims is for access, as opposed to other use cases such as refresh. And checking the `sub` exists is for subsequent checks in other files that validate that the subject of the claims is for B2B (validating the URN)

Reply:
I moved the function out because I also needed to get the validated claims when doing Platform authentication. Because the token sent to B2B is a JWT, as opposed to an access token, the Platform code in `authentication.py` throws and invalid access token error when we hit the Platform introspect route. So to get around this, I allow the request to be authenticated if the token is a valid Platform access token, or if the token is a valid Platform JWT.